app/main.py

```diff
@@ -1,2 +1,94 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = self.create_hash_table()
+
+    def __setitem__(self, key, value):
+        threshold = int(self.load_factor * self.capacity)
+        if self.size == threshold:
+            self.resize()
+        hash_value = hash(key)
+        self.form_hash_table(key, value, hash_value)
+
+    def form_hash_table(self, key, value, hash_value):
+        index = hash_value % self.capacity
+
+        while True:
+            if not self.hash_table[index]:
+                self.size += 1
+                self.hash_table[index] = [key, value, hash_value]
+                break
+            if self.hash_table[index][2] == hash_value \
+                    and self.hash_table[index][0] == key:
+                self.hash_table[index][1] = value
+                break
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key):
+        hash_value = hash(key)
+        index = hash_value % self.capacity
+
+        while self.hash_table[index]:
+            if self.hash_table[index][2] == hash_value \
+                    and self.hash_table[index][0] == key:
+                return self.hash_table[index][1]
+            index = (index + 1) % self.capacity
+        raise KeyError(key)
+
+    def __len__(self):
+        return self.size
+
+    def resize(self):
+        self.capacity *= 2
+        copied_elements = self.hash_table.copy()
+        self.hash_table = self.create_hash_table()
+        self.size = 0
+        for element in copied_elements:
+            if element:
+                self.__setitem__(element[0], element[1])
+
+    def create_hash_table(self):
+        return [[] for _ in range(self.capacity)]
+
+    def clear(self):
+        self.hash_table = self.create_hash_table()
+        return self
+
+    def __delitem__(self, key):
+        for ind, val in enumerate(self.hash_table):
```

Comment:
I just wanted to praise you, but here you used abbreviations for names
Abbreviations are appropriate for long names, in this case, you shortened by 2-3 letters, so is it significant? It is better to complete the word completely, it will be more readable